pyoverkiz/client.py

```diff
@@ -508,25 +601,45 @@ async def execute_scenario(self, oid: str) -> str:
 
     async def __get(self, path: str) -> Any:
         """Make a GET request to the OverKiz API"""
-        async with self.session.get(f"{self.server.endpoint}{path}") as response:
+        headers = {}
+
+        if self.server == SUPPORTED_SERVERS["somfy_europe"]:
+            await self._refresh_somfy_tahoma_token_if_expired()
+            headers["Authorization"] = f"Bearer {self._access_token}"
+
+        async with self.session.get(
+            f"{self.server.endpoint}{path}", headers=headers
+        ) as response:
             await self.check_response(response)
             return await response.json()
 
     async def __post(
         self, path: str, payload: JSON | None = None, data: JSON | None = None
     ) -> Any:
         """Make a POST request to the OverKiz API"""
+        headers = {}
+
+        if self.server == SUPPORTED_SERVERS["somfy_europe"] and path != "login":
+            await self._refresh_somfy_tahoma_token_if_expired()
+            headers["Authorization"] = f"Bearer {self._access_token}"
+
         async with self.session.post(
-            f"{self.server.endpoint}{path}",
-            data=data,
-            json=payload,
+            f"{self.server.endpoint}{path}", data=data, json=payload, headers=headers
         ) as response:
             await self.check_response(response)
             return await response.json()
 
     async def __delete(self, path: str) -> None:
         """Make a DELETE request to the OverKiz API"""
-        async with self.session.delete(f"{self.server.endpoint}{path}") as response:
+        headers = {}
+
+        if self.server == SUPPORTED_SERVERS["somfy_europe"]:
+            await self._refresh_somfy_tahoma_token_if_expired()
+            headers["Authorization"] = f"Bearer {self._access_token}"
```

Comment:
Can put somewhere else later to avoid dup code.

Reply:
I thought about that one, and pass the headers to that function, however for now I found this clear enough.

Would this be ok for this current fix?